Добавление методов взаимодействия с блоками в игровом пространстве

Irautak opened the pull request.

Shaniser committed:
Added GameArea.js (not complited)

Irautak committed:
Добавление методов взаимодействия с блоками в игровое пространство

Добавление методов взаимодействия с блоками в игровое пространство.
Добавление точек с запятой в код, в соответствии со стилем.

Irautak committed:
Revert "Добавление методов взаимодействия с блоками в игровое пространство"

This reverts commit 5aa9fce627d2338d1473f5d1b668050a7b305295.

Irautak committed:
Добавление методов взаимодействия с блоками в игровом пространстве

Mitbus commented on Scripts/GameArea.js:
```diff
@@ -0,0 +1,126 @@
+		// Проверка окружения блока, при наличии у него особого поведения. Особое поведение определеяется по типу
+		this.makeAirBlock = (coordinates) => {
+			// Делает блок воздуха
+			return new Block(0, "air", coordinates, 100, 0,
```
Давайте использовать не блоки воздуха, а просто undefined (ничего). Так в случае необходимости мы сможем вводить прозрачные для игрока блоки + не придется отрисовывать лишние блоки, когда их нет

Mitbus commented on Scripts/GameArea.js:
```diff
@@ -0,0 +1,126 @@
+
+// Класс блока
+class Block{
+	constructor(id, type, coordinates, durability, brightness, isCollissed, isPlatform, isClickable){
```
Добавьте описание сложных полей блока (не понятно, что такое id и type, например)

Mitbus commented on Scripts/GameArea.js:
```diff
@@ -0,0 +1,126 @@
+		};
+
+		// Обновление окружения блока
+		this.updateRadius = (coordinates) => {
```
Можно вызывать не каскад функций выше, а одну в цикле с разными параметрами? Копипаст - это плохо

Mitbus commented on Scripts/GameArea.js:
```diff
@@ -0,0 +1,126 @@
+				false, false, false);
+		};
+
+		// =====Группа функций, возвращающая координаты относительно заданной=====
```
Можно объединить

Irautak committed:
Фикс представления map + добавление тривиального поведения воды и деревьев

1) Теперь map хранит лишь id поставленных блоков, описания типов блоков
лежит в переменной block_table
2) Отказ от класса Coordinate
3) Добавление описания характеристик блоков
4) Добавления тривиального поведения воды
5) Добавления поведения деревьев (дерево падает, если не имеет опоры)
